File: Easy/0168_excel-sheet-column-title/excel-sheet-column-title.go
package main

import (
	"fmt"
)

func convertToTitle(columnNumber int) string {
	result := make([]byte, 0)
	for columnNumber > 0 {
		columnNumber--
		result = append(result, byte('A' + columnNumber % 26))
		columnNumber /= 26
	}

	for i, j := 0, len(result)-1; i < j; i, j = i+1, j-1 {
		result[i], result[j] = result[j], result[i]
	}
  return string(result)
}

func main() {
	tests := []int{
		1,
		28,
		701,
	}

	for _, n := range tests {
		fmt.Printf("---\nNumber: %d Title: %s\n", n, convertToTitle(n))
	}
}

File: Easy/0168_excel-sheet-column-title/excel-sheet-column-title.py
#!/bin/python3

from typing import List

class Solution:
    def convertToTitle(self, columnNumber: int) -> str:
        result = ""
        while columnNumber > 0:
            columnNumber -= 1
            result = chr(ord('A') + columnNumber % 26) + result
            columnNumber //= 26
        return result

def main():
    tests : List[int] = [
        1,
        28,
        701,
    ]
    for n in tests:
        print(f"---\nNumber: {n} Title: {Solution().convertToTitle(n)}")

if __name__ == "__main__":
    main()
